src/bitcoin-cli.cpp

```diff
@@ -55,7 +55,10 @@ static constexpr int DEFAULT_WAIT_CLIENT_TIMEOUT = 0;
 static const bool DEFAULT_NAMED=false;
 static const int CONTINUE_EXECUTION=-1;
 static constexpr int8_t UNKNOWN_NETWORK{-1};
-static constexpr std::array NETWORKS{"ipv4", "ipv6", "onion", "i2p", "cjdns"};
+// See GetNetworkName() in netbase.cpp
+static constexpr std::array NETWORKS{"not_publicly_routable", "ipv4", "ipv6", "onion", "i2p", "cjdns", "internal"};
+static constexpr std::array NETWORK_NAMES{"npr", "ipv4", "ipv6", "onion", "i2p", "cjdns", "int"};
+static constexpr std::array NETWORK_UNREACHABLE{0 /* "not_publicly_routable" */, 6 /* "internal" */};
```

Comment:
This code looks like it should probably get some refactoring. No reason it can't work with network strings (not ids) and figure out what "unlisted" networks have peers on its own, and doing so would be more future-proof as an added bonus.

Reply:
Good idea -- would it make sense to `#include <netbase.h>` in `bitcoin-cli.cpp`? Then we can share one single list `enum Network` as well as `GetNetworkName()` and `ParseNetwork()` without re-writing those in bitcoin-cli. I could also add a new function `GetNetworkShortName()` in `netbase.cpp` so everything is in one place. If new networks are added in the future, `bitcoin-cli.cpp` shouldn't need any changes...

Reply:
@luke-jr gave this refactor a try in 7dd8b095ada2711487057788206deb7e310b8092

Reply:
No, I'm saying there shouldn't be an `enum Network` or `network_id` at all - just use the strings.

Reply:
ok something like `std::map<std::string, int>` to keep track of count of peers by network string?